Source/Evolve/WarpXEvolve.cpp

```diff
@@ -488,15 +488,6 @@ void WarpX::HandleParticlesAtBoundaries (int step, amrex::Real cur_time, int num
 {
     mypc->ContinuousFluxInjection(cur_time, dt[0]);
 
-    mypc->ApplyBoundaryConditions();
```

Comment:
Potentially, adding a test similar as in current deposition, that checks travel distance is needed here (for EB).
Essentially, particles that shall undergo EB handling need to be checked if they are within the ngrow of the iMultiFab of the EBs.

Reply:
The problem I see is if a particles passes through an EB and then goes outside the domain. I don't see an easy way to deal with this case, associating that lost particle to the EB instead of the domain boundary. It would be even worse if the particle went beyond any guard cells.